lib/active_shipping/tracking_response.rb

```diff
@@ -101,5 +101,21 @@ def has_exception?
     alias_method :scheduled_delivery_time, :scheduled_delivery_date
     alias_method :actual_delivery_time, :actual_delivery_date
     alias_method :attempted_delivery_time, :attempted_delivery_date
+
+    def ==(other)
+      carrier == other.carrier && carrier_name == other.carrier_name && status == other.status &&
+        status_description == other.status_description && ship_time == other.ship_time &&
+        scheduled_delivery_date == other.scheduled_delivery_date &&
+        actual_delivery_date == other.actual_delivery_date &&
+        attempted_delivery_date == other.attempted_delivery_date && delivery_signature == other.delivery_signature &&
+        tracking_number == other.tracking_number && compare_shipment_events(other) &&
+        shipper_address == other.shipper_address && origin == other.origin && destination == other.destination
+    end
+
+    private
+    # Returns true if shipment events are the same
+    def compare_shipment_events(other)
+      shipment_events.size == other.shipment_events.size && shipment_events & other.shipment_events == shipment_events
```

Comment:
`shipment_events & other.shipment_events == shipment_events`

noice.

Like jon said though you should break up the lines above and use iteration, easier to read.